packages/manager/modules/hub/src/components/products/controller.js

```diff
@@ -4,17 +4,52 @@ import {
 } from './constants';
 
 export default class ProductsController {
-  constructor() {
+  /* @ngInject */
+  constructor($state, atInternet) {
     this.DEFAULT_DISPLAYED_TILES = DEFAULT_DISPLAYED_TILES;
     this.DISPLAYED_PRODUCTS_NUMBER = DISPLAYED_PRODUCTS_NUMBER;
+    this.$state = $state;
+    this.atInternet = atInternet;
+    this.loading = true;
+  }
+
+  $onInit() {
+    this.skeletonServices = Array.from({ length: 6 });
+    this.skeletonProducts = Array.from({ length: 4 });
+    this.products
```

Comment:
it looks unclear to use a promise for "products" component binding, it makes more sense to expect a list of products

Reply:
I agree. 

Though this is the result of using the `resolvePolicy : NOWAIT` in the caller component. 

I think it would be tricky to pass the products directly to the component bindings because the caller has no controller where we can handle the loading states, but i can actually rename the binding so that products would be "products-promise" or something like that to make it clear that it's not the products, what do you think?

Reply:
yes something like ```productsPromise``` would be better suited :)